ros2doctor/ros2doctor/api/network.py

```diff
@@ -74,11 +76,10 @@ def check(self):
         if not _is_unix_like_platform():
             if not has_loopback and not has_non_loopback:
                 # no flags found, otherwise one of them should be True.
-                doctor_error(
-                    'No flags found. '
-                    'Run `ipconfig` on Windows or '
-                    'install `ifconfig` on Unix to check network interfaces.')
-                result.add_error()
+                doctor_warn(
+                    'Did not found either a loopback or no loopback interface.'
+                    'Run `ipconfig` on Windows to check what interfaces are available')
```

Comment:
> Separately, why is there no similar warning for Unix-like platforms?

I think that on Unix if you get that output here you actually don't have a network interface.
On windows this check sometimes fails when you have a network interface available, because the `ifcfg` package doesn't populate the interface "flags" when listing interfaces.

I applied your suggestion though actually the issue isn't than an interface wasn't found, but that we cannot recognize which one is a loopback interface and which one not.

See also https://github.com/ros2/ros2cli/pull/404.

Reply:
Modified message again, see latest changes.